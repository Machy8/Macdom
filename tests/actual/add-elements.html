svg $100 $100 Inner text
elementxy $Some data content;