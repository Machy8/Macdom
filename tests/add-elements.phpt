<?php

use Tester\Assert;


require __DIR__ . '/bootstrap.php';

$macdom->setup->addElements = [
	'svg' => [
		'qkAttributes' => ['width', 'height']
	],

	'elementxy' => [
		'unpaired',
		'qkAttributes' => ['data-somedata']
	]
];

$macdom->setup->compressText = TRUE;

Assert::matchFile(
	EXPECTED . "/add-elements.html",
	$macdom->compile(file_get_contents(ACTUAL . "/add-elements.html"))
);
