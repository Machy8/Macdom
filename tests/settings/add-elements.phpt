<?php

use Tester\Assert;

require '../bootstrap.php';

$o->addElements([
	'svg' => [
      'qkAttributes' => ['width', 'height']
   ],
	'elementxy' => [
		'unpaired',
		'qkAttributes' => ['data-somedata']
	]
]);

Assert::same('<svg width="100" height="100">Inner text</svg>', $o->compileContent("svg $100 $100 Inner text"));
Assert::same('<elementxy data-somedata="Some data content" />', $o->compileContent('elementxy $Some data content;'));