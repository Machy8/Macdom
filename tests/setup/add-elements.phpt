<?php

use Tester\Assert;

require '../bootstrap.php';

$testedA = 'svg $100 $100 Inner text';
$resultA = '<svg width="100" height="100">Inner text</svg>';

$testedB = 'elementxy $Some data content;';
$resultB = '<elementxy data-somedata="Some data content">';

$o->setup->addElements = [
	'svg' => [
		'qkAttributes' => ['width', 'height']
	],
	'elementxy' => [
		'unpaired',
		'qkAttributes' => ['data-somedata']
	]
];

$o->setup->compressCode = TRUE;

Assert::same($resultA, $o->compile($testedA));
Assert::same($resultB, $o->compile($testedB));
