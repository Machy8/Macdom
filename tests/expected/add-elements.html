<svg width="100" height="100">Inner text</svg>
<elementxy data-somedata="Some data content">